### PR TITLE
TeamCity: Refactor config tests

### PR DESCRIPTION
**Pull request by SarahFrench**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
I'm going to keep this as draft until the GHA implemented in https://github.com/GoogleCloudPlatform/magic-modules/pull/9954 is available to help reviewers

**Comment by SarahFrench** [comment]
Rebased to include https://github.com/GoogleCloudPlatform/magic-modules/pull/9954

File: .github/workflows/teamcity-pr-checks.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
   pull_request:
     paths:
       - '.github/workflows/teamcity-pr-checks.yml'
-      - 'mmv1/third_party/terraform/!.teamcity'
+      - 'mmv1/third_party/terraform/.teamcity'
```
Comment (SarahFrench) [inline .github/workflows/teamcity-pr-checks.yml:9]: I thought that the `!` was required to use a folder starting with a dot but looks like I was wrong!

**Comment by SarahFrench** [comment]
I'm going to wait for https://github.com/GoogleCloudPlatform/magic-modules/pull/10003 to be merged and rebase again 🤦

**Comment by SarahFrench** [comment]
Yay, after the last rebase the TeamCity config check is running as expected 🎉

**Review by BBBmau — APPROVED** [review]

**Comment by SarahFrench** [comment]
Rebased to include https://github.com/GoogleCloudPlatform/magic-modules/pull/10025, addressed merge conflicts and used this PR's work to refactor the new tests added

**Pull request merged**